Graph/TestsForGraph/runTests.sh

```diff
@@ -0,0 +1,72 @@
+#!/bin/bash
+
+argumentForCheckingFails=0
+existingOfMake=0
+
+cd ../
+if [[ -f "./minDist" ]]
+then
+    existingOfMake=1
+fi
+
+make > /dev/null
+
+./minDist ./TestsForGraph/test1.txt > testForCheck.txt
+
+if [ "$echo $(cat ./TestsForGraph/test1_output.txt)" != "$echo $(cat testForCheck.txt)" ]
+then
+	echo "Failed when running test1"
+	argumentForCheckingFails=1
+fi
+
+rm testForCheck.txt
+
+cd TestsForGraph
+g++ -o forInput testOfCircleGraph.cpp
```

Comment:
Your testing script must only test the project, no build the test. separate building parts. You can create a Makefile for building necessary binaries for testing